src/map.h

```diff
@@ -17,36 +17,32 @@ struct FindPathParams;
 struct AStarNode
 {
 	AStarNode* parent;
-	int_fast32_t f;
+	double g;
+	double f;
 	uint16_t x, y;
 };
 
-static constexpr int32_t MAX_NODES = 512;
-
-static constexpr int32_t MAP_NORMALWALKCOST = 10;
-static constexpr int32_t MAP_DIAGONALWALKCOST = 25;
+static constexpr float MAP_NORMALWALKCOST = 1.0f;
+static constexpr float MAP_DIAGONALWALKCOST = 2.5f;
 
 class AStarNodes
 {
 public:
-	AStarNodes(uint32_t x, uint32_t y);
+	AStarNodes(uint16_t x, uint16_t y);
+
+	void createNewNode(AStarNode* parent, uint16_t x, uint16_t y, double g, double f);
+	void addNode(AStarNode* node) { nodes.push_back(node); };
 
-	AStarNode* createOpenNode(AStarNode* parent, uint32_t x, uint32_t y, int_fast32_t f);
 	AStarNode* getBestNode();
-	void closeNode(AStarNode* node);
-	void openNode(AStarNode* node);
-	int_fast32_t getClosedNodes() const;
-	AStarNode* getNodeByPosition(uint32_t x, uint32_t y);
+	AStarNode* getNodeByPosition(uint16_t x, uint16_t y) { return nodeMap[x][y]; };
 
-	static int_fast32_t getMapWalkCost(AStarNode* node, const Position& neighborPos);
-	static int_fast32_t getTileWalkCost(const Creature& creature, const Tile* tile);
+	static double calculateHeuristic(const Position& p1, const Position& p2);
```

Comment:
I would prefer a bit more descriptive name for this function, but it's not really a deal breaker tbh.

Reply:
Maybe we can make up a name together. If you learn a bit about pathfinding algorithms that value is actually called the heuristic so I figured naming things as they are seen when learning about pathfinding would be the best way.

Reply:
chatGPT:
calculateDistance
computeDistance
estimateDistance
evaluateDistance
findDistance
computeHypotenuse
estimateHypotenuse
computeEuclideanDistance
evaluateEuclideanDistance
computeTriangleHypotenuse
estimateTriangleHypotenuse
calculateTriangleDistance
estimateTriangleDistance
computePositionDistance
estimatePositionDistance

Reply:
I think the name I chose beats these lol

Reply:
findShortcut()
shortestPath()
fastRoute()
quickLine()

These aren't that great either, but not bad.

Reply:
Maybe something like distanceToEnd() or getStraightLineToEnd()

Reply:
This is usually just called `h` in most papers and implementations anyway. `estimateCost` is fine, it's not supposed to be too descriptive because we can change the function at our discretion